storage/remote/shards.go

```diff
@@ -0,0 +1,289 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
```

Comment:
Lets avoid doing a big code move in this PR; it makes it harder to review.

Reply:
I tried putting in a separate commit to make it a bit easier. I find that the `metaShards` might be a bit difficult to read w/o it.

I'll remove the parallelization and place the sharding code back to the same file.